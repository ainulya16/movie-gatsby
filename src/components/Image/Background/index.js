import React from 'react';
import PropTypes from 'prop-types';
import { StaticQuery, graphql } from 'gatsby';
import BackgroundImageGatsby from 'gatsby-background-image';
import styled from 'styled-components';

const BackgroundImageComponent = props => {
  const { filename, alt, backgroundColor, children, ...attr } = props;

  return (
    <StaticQuery
      query={graphql`
        query {
          images: allFile {
            edges {
              node {
                relativePath
                name
                childImageSharp {
                  fluid(maxWidth: 4160, quality: 100) {
                    ...GatsbyImageSharpFluid_withWebp
                  }
                }
              }
            }
          }
        }
      `}
      render={data => {
        const image = data.images.edges.find(n => {
          return n.node.relativePath.includes(filename);
        });
        if (!image) return null;
        const imageData = image.node.childImageSharp.fluid;
        return (
          <BackgroundImageGatsby
            alt={alt}
            fluid={imageData}
            backgroundColor={backgroundColor}
            {...attr}
          >
            {children}
          </BackgroundImageGatsby>
        );
      }}
    />
  );
};

BackgroundImageComponent.defaultProps = {
  filename: 'blazing.png',
  alt: 'logo',
  backgroundColor: '#FFF',
  children: null,
};

BackgroundImageComponent.propTypes = {
  filename: PropTypes.string,
  alt: PropTypes.string,
  backgroundColor: PropTypes.string,
  children: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.arrayOf(PropTypes.node),
  ]),
};
const BackgroundImage = styled(BackgroundImageComponent)`
  width: 100%;
  height: 500px;
  background-position: bottom center;
  background-repeat: repeat-y;
  background-size: cover;
`;

BackgroundImage.displayName = 'BackgroundImage';

export default BackgroundImage;
