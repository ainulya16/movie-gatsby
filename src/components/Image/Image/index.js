import React from 'react';
import PropTypes from 'prop-types';
import { StaticQuery, graphql } from 'gatsby';
import Img from 'gatsby-image/withIEPolyfill';

const Image = props => {
  const { filename, alt, ...attr } = props;
  return (
    <StaticQuery
      query={graphql`
        query {
          images: allFile {
            edges {
              node {
                relativePath
                name
                childImageSharp {
                  sizes(maxWidth: 600, quality: 100) {
                    ...GatsbyImageSharpSizes
                  }
                }
              }
            }
          }
        }
      `}
      render={data => {
        const image = data.images.edges.find(n => {
          return n.node.relativePath.includes(filename);
        });
        if (!image) return null;
        const imageSizes = image.node.childImageSharp.sizes;
        return (
          <Img
            alt={alt}
            sizes={imageSizes}
            objectFit="cover"
            objectPosition="50% 50%"
            {...attr}
          />
        );
      }}
    />
  );
};

Image.defaultProps = {
  filename: 'aia-logo.png',
  alt: 'logo',
};

Image.propTypes = {
  filename: PropTypes.string,
  alt: PropTypes.string,
};

export default Image;
