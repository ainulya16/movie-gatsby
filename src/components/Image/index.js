import BackgroundImage from './Background';
import Image from './Image';

export { BackgroundImage, Image };
