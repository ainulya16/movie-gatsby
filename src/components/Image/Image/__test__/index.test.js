import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import { StaticQuery } from 'gatsby';
import Image from '..';
// images: allFile {
//   edges {
//     node {
//       relativePath
//       name
//       childImageSharp {
//         sizes(maxWidth: 600, quality:100) {
//           ...GatsbyImageSharpSizes
//         }
//       }
//     }
//   }
// }
beforeEach(() => {
  StaticQuery.mockImplementationOnce(({ render }) =>
    render({
      images: {
        edges: [
          {
            node: {
              relativePath: 'aia-logo-white.png',
              name: 'aia-logo-white',
              childImageSharp: {
                sizes: {
                  base64:
                    'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAAUCAYAAACNiR0NAAAACXBIWXMAAAsSAAALEgHS3X78AAACZ0lEQVQ4y32UW4iNURTHv3MxTMYllzInjTGKya14EEbRpLyQ2+BpxqQhJbyNNDIuk9uLURIvrg8oClFEFPJizCA8kLxRSinyIJ3jt/RfxzpfsuvXXmvvtda391prf0mpVEps2Ay5RAN5LVyEPngPb+ExnIB5wS7n/sVisbxoSlZyMzyT825YAjNhNrTAMXgF16CQDuoBPdh2eA1Lk/8MCwBd8A7mxKAx+gZO+YC5WnoG8nLOeDBshoQ0zYeX0Ki9rG80wl0YLD1v+fBAzBZkql9Ra267CG6kr2B5afZgKpDvFWAs1PCRA8ydME57fptdsN4d6qE3BM8HeRZMsVm6FcjGJ6gNdgU+1uvKclihkzVobSiswmg18wVoDXl9qKD3YUIIugfqTGiFzTAadsJlAlm1J0MVnPITynFb6e/4YdfVusWZa0K35VCL1grn/3ypMsc9dmLJdfC9VDmsJ59Y1c1gC7SZA9zkdFag+lRL5XXVaWqPNXAE/Vwq8CTvJWuZr1q8BDOUr5xaah98sfZIPbse+TzV66qxxVpot76KlQtOTdpvgOGW17C3UQcaBR3RaR2MD/om5aRJ+giuOEzyVnhk7zvYL4QFMaAleqXk4/AZBuC2ctYXqvkN/Z4qvExr7fHZ+Y+hGu5AUUWwFrquvefWZ5I/6gAtyl9nOFjGH/wYlDcy/gCH4QUclOEANnslW/LPol/VP/InnEz/vizZO2AkTIdb9mmc2rTfD/v18SsWhPkoVKloHeWA5aiVlR0Ev2CxXxnnQ5LPWMH+4VNxwiT+9/QMT8NE6VaELsn2CLrD286GOiS/Aa/iv8rymE3HAAAAAElFTkSuQmCC',
                  aspectRatio: 1,
                  src:
                    '/static/626b8c41be4495457c39299e8dab6e35/c0051/aia-logo-white.png',
                  srcSet:
                    '/static/626b8c41be4495457c39299e8dab6e35/c0051/aia-logo-white.png 80w',
                  srcWebp:
                    '/static/626b8c41be4495457c39299e8dab6e35/ab258/aia-logo-white.webp',
                  srcSetWebp:
                    '/static/626b8c41be4495457c39299e8dab6e35/ab258/aia-logo-white.webp 80w',
                  sizes: '(max-width: 80px) 100vw, 80px',
                },
              },
            },
          },
        ],
      },
    }),
  );
});

describe('Image Component', () => {
  it('should be exist', () => {
    const component = shallow(<Image filename="aia-logo-white.png" />);
    expect(component.exists()).toBe(true);
  });
  it('should match snapshot', () => {
    const tree = renderer
      .create(<Image filename="aia-logo-white.png" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should match snapshot with no exist filename', () => {
    const tree = renderer
      .create(<Image filename="aia-logo-white-not-exist.png" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
