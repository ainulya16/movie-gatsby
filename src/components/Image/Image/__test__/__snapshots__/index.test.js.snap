// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Component should match snapshot 1`] = `
<div
  className=" gatsby-image-wrapper"
  style={
    Object {
      "overflow": "hidden",
      "position": "relative",
    }
  }
>
  <div
    style={
      Object {
        "paddingBottom": "100%",
        "width": "100%",
      }
    }
  />
  <img
    alt=""
    src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABQAAAAUCAYAAACNiR0NAAAACXBIWXMAAAsSAAALEgHS3X78AAACZ0lEQVQ4y32UW4iNURTHv3MxTMYllzInjTGKya14EEbRpLyQ2+BpxqQhJbyNNDIuk9uLURIvrg8oClFEFPJizCA8kLxRSinyIJ3jt/RfxzpfsuvXXmvvtda391prf0mpVEps2Ay5RAN5LVyEPngPb+ExnIB5wS7n/sVisbxoSlZyMzyT825YAjNhNrTAMXgF16CQDuoBPdh2eA1Lk/8MCwBd8A7mxKAx+gZO+YC5WnoG8nLOeDBshoQ0zYeX0Ki9rG80wl0YLD1v+fBAzBZkql9Ra267CG6kr2B5afZgKpDvFWAs1PCRA8ydME57fptdsN4d6qE3BM8HeRZMsVm6FcjGJ6gNdgU+1uvKclihkzVobSiswmg18wVoDXl9qKD3YUIIugfqTGiFzTAadsJlAlm1J0MVnPITynFb6e/4YdfVusWZa0K35VCL1grn/3ypMsc9dmLJdfC9VDmsJ59Y1c1gC7SZA9zkdFag+lRL5XXVaWqPNXAE/Vwq8CTvJWuZr1q8BDOUr5xaah98sfZIPbse+TzV66qxxVpot76KlQtOTdpvgOGW17C3UQcaBR3RaR2MD/om5aRJ+giuOEzyVnhk7zvYL4QFMaAleqXk4/AZBuC2ctYXqvkN/Z4qvExr7fHZ+Y+hGu5AUUWwFrquvefWZ5I/6gAtyl9nOFjGH/wYlDcy/gCH4QUclOEANnslW/LPol/VP/InnEz/vizZO2AkTIdb9mmc2rTfD/v18SsWhPkoVKloHeWA5aiVlR0Ev2CxXxnnQ5LPWMH+4VNxwiT+9/QMT8NE6VaELsn2CLrD286GOiS/Aa/iv8rymE3HAAAAAElFTkSuQmCC"
    style={
      Object {
        "fontFamily": "\\"object-fit: cover; object-position: 50% 50%\\"",
        "height": "100%",
        "left": 0,
        "objectFit": "cover",
        "objectPosition": "50% 50%",
        "opacity": 1,
        "position": "absolute",
        "top": 0,
        "transitionDelay": "500ms",
        "width": "100%",
      }
    }
  />
  <picture>
    <source
      sizes="(max-width: 80px) 100vw, 80px"
      srcSet="/static/626b8c41be4495457c39299e8dab6e35/ab258/aia-logo-white.webp 80w"
      type="image/webp"
    />
    <source
      sizes="(max-width: 80px) 100vw, 80px"
      srcSet="/static/626b8c41be4495457c39299e8dab6e35/c0051/aia-logo-white.png 80w"
    />
    <img
      alt="logo"
      loading="lazy"
      onLoad={[Function]}
      sizes="(max-width: 80px) 100vw, 80px"
      src="/static/626b8c41be4495457c39299e8dab6e35/c0051/aia-logo-white.png"
      srcSet="/static/626b8c41be4495457c39299e8dab6e35/c0051/aia-logo-white.png 80w"
      style={
        Object {
          "fontFamily": "\\"object-fit: cover; object-position: 50% 50%\\"",
          "height": "100%",
          "left": 0,
          "objectFit": "cover",
          "objectPosition": "50% 50%",
          "opacity": 0,
          "position": "absolute",
          "top": 0,
          "transition": "opacity 500ms",
          "width": "100%",
        }
      }
    />
  </picture>
  <noscript
    dangerouslySetInnerHTML={
      Object {
        "__html": "<picture><source type='image/webp' srcset=\\"/static/626b8c41be4495457c39299e8dab6e35/ab258/aia-logo-white.webp 80w\\" sizes=\\"(max-width: 80px) 100vw, 80px\\" /><source srcset=\\"/static/626b8c41be4495457c39299e8dab6e35/c0051/aia-logo-white.png 80w\\" sizes=\\"(max-width: 80px) 100vw, 80px\\" /><img loading=\\"lazy\\" sizes=\\"(max-width: 80px) 100vw, 80px\\" srcset=\\"/static/626b8c41be4495457c39299e8dab6e35/c0051/aia-logo-white.png 80w\\" src=\\"/static/626b8c41be4495457c39299e8dab6e35/c0051/aia-logo-white.png\\" alt=\\"logo\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
      }
    }
  />
</div>
`;

exports[`Image Component should match snapshot with no exist filename 1`] = `null`;
